### DISCUSSION TITLE
Gatsby GraphQL Error

### DISCUSSION THREAD

**Discussion by fjshomeplate** [gatsbyjs/gatsby#28462] [Help] [answered]
I have been trying to source content through Graphql for a store site, and source the data into a component. I have been getting this error when I go to the site on localhost: 
Error: It appears like Gatsby is misconfigured. Gatsby related `graphql` calls are supposed to only be evaluated at compile time, and then compiled away. Unfortunately, something went wrong and the query was left in the compiled code.

Unless your site has a complex or custom babel/Gatsby configuration this is likely a bug in Gatsby.

They only babel configuration I have done is downgrade babel-remove-graphql-queries, but after I deleted node_modules well as deleting package-lock.json, it doesn't change. Any ideas?

**Comment by ascorbic** [comment, marked as answer]
Hi. That error means that `babel-remove-graphql-queries` isn't being run. Why did you downgrade it?

**Reply by fjshomeplate** [reply]
I downgraded when trying to fix Static Query(Loading...), which I fixed a different way

**Reply by ascorbic** [reply]
If you upgrade everything, does it work?

**Reply by fjshomeplate** [reply]
Not Sure. Let me try.

**Reply by fjshomeplate** [reply]
I updated it, and unfortunately, it didn't work.

**Reply by ascorbic** [reply]
Can you do a minimal reproduction?

**Reply by fjshomeplate** [reply]
What do you mean?

**Reply by ascorbic** [reply]
This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it. See how to create one here: https://gatsby.dev/reproduction

**Reply by fjshomeplate** [reply]
The repo is here: https://github.com/fjshomeplate/gatsby-bug-ql.

**Reply by vladar** [reply]
Thanks for the repro.

But what are the steps to actually reproduce it with this repo? I've run `gatsby develop` and on the first run it showed this error:

```
error "gatsby-source-graphql" threw an error while running the sourceNodes lifecycle:

only absolute urls are supported
```

That's because you have `gatsby-source-graphql` in site plugins without the actual endpoint: https://github.com/fjshomeplate/gatsby-bug-ql/blob/8466636449dc0cf9d040b1cb068f691094e44854/gatsby-config.js#L7-L16

I've removed this plugin and re-ran `gatsby develop`. It built the site but I don't see the error you mention on any of the pages. Where should I look to see it?

**Reply by fjshomeplate** [reply]
try downgrading babel-remove-queries

**Comment by fjshomeplate** [comment]
I didn't really fix it, just migrates the code. :( Oh, well. Didn't have much in there anyway.
